test/priors/test_lkj_prior.py

```diff
@@ -184,19 +191,31 @@ def test_lkj_covariance_prior_batch_log_prob(self, cuda=False):
         v = torch.ones(2, 1, device=device)
         sd_prior = SmoothedBoxPrior(exp(-1) * v, exp(1) * v)
         prior = LKJCovariancePrior(2, torch.tensor([0.5, 1.5], device=device), sd_prior)
+        corr_dist = LKJCholesky(2, torch.tensor([0.5, 1.5], device=device))
 
         S = torch.eye(2, device=device)
-        self.assertTrue(approx_equal(prior.log_prob(S), torch.tensor([-3.59981, -2.21351], device=S.device)))
+        dist_log_prob = corr_dist.log_prob(S) + sd_prior.log_prob(S.diag())
+        self.assertLessEqual((prior.log_prob(S) - dist_log_prob).abs().sum(), 1e-4)
+
         S = torch.stack([S, torch.tensor([[1.0, 0.5], [0.5, 1]], device=S.device)])
-        self.assertTrue(approx_equal(prior.log_prob(S), torch.tensor([-3.59981, -2.35735], device=S.device)))
-        with self.assertRaises(ValueError):
-            prior.log_prob(torch.eye(3, device=device))
+        S_chol = torch.linalg.cholesky(S)
+        dist_log_prob = corr_dist.log_prob(S_chol) + sd_prior.log_prob(torch.diagonal(S, dim1=-2, dim2=-1))
+        self.assertLessEqual((prior.log_prob(S) - dist_log_prob).abs().sum(), 1e-4)
 
     def test_lkj_covariance_prior_batch_log_prob_cuda(self):
         if torch.cuda.is_available():
             with least_used_cuda_device():
                 self.test_lkj_covariance_prior_batch_log_prob(cuda=True)
 
+    def test_lkj_prior_sample(self):
+        prior = LKJCovariancePrior(2, 0.5, sd_prior=SmoothedBoxPrior(exp(-1), exp(1)))
+        random_samples = prior.sample(torch.Size((6,)))
+        # only need to check that this is a PSD matrix
```

Comment:
Did you want to check symmetry as well? I think `eigh` will produce (wrong) results even if the matrix is not symmetric

Reply:
Everything ought to be symmetric by construction, but probably a good idea.

Reply:
Yup, but the construction happens in your code :)